hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenUpdateValidator.java

```diff
@@ -56,12 +56,13 @@ public ValidationResult validateSemantics(
         final var tokenId = op.tokenOrThrow();
         final var token = getIfUsable(tokenId, tokenStore);
         final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
-        // If the token has an empty admin key it can't be updated for any other fields other than metadata
-        // For updating only metadata the transaction should have admin key or metadata key
-        if (isMetadataOnlyUpdateOp(op)) {
-            validateTrue(token.hasAdminKey() || token.hasMetadataKey(), TOKEN_IS_IMMUTABLE);
-        } else if (!isExpiryOnlyUpdateOp(op)) {
-            validateFalse(isEmpty(token.adminKey()), TOKEN_IS_IMMUTABLE);
+        // If the token has an empty admin key it can't be updated for any other fields other than low priority keys or
+        // expiry
+        // For updating only low priority keys - (wipeKey, kycKey, supplyKey, freezeKey, feeScheduleKey, pauseKey or
+        // metadataKey)
+        // the transaction should have admin key or the respective low priority key signature
+        if (isEmpty(token.adminKey())) {
+            validateTrue(isExpiryOnlyUpdateOp(op) || hasAlreadySomeNonAdminKeys(token), TOKEN_IS_IMMUTABLE);
```

Comment:
How could we be sure that the key we are trying to modify is present using `hasAlreadySomeNonAdminKeys` ?

I think this should be taking the transaction body and validating the key we are trying to modify is present, otherwise throw `TOKEN_IS_IMMMUTABLE`